src/main/java/spoon/reflect/visitor/DefaultJavaPrettyPrinter.java

```diff
@@ -742,7 +743,12 @@ public void visitCtDo(CtDo doLoop) {
 	@Override
 	public <T extends Enum<?>> void visitCtEnum(CtEnum<T> ctEnum) {
 		visitCtType(ctEnum);
-		printer.writeKeyword("enum").writeSpace().writeIdentifier(ctEnum.getSimpleName());
+		printer.writeKeyword("enum").writeSpace();
+		if (ctEnum.isLocalType()) {
+			printer.writeIdentifier(stripLeadingDigits(ctEnum.getSimpleName()));
+		} else {
+			printer.writeIdentifier(ctEnum.getSimpleName());
+		}
```

Comment:
I just have one thought, and that's if we should simplify this like so:

```suggestion
		printer.writeIdentifier(stripLeadingDigits(ctEnum.getSimpleName()));
```

This should effectively provide the same behavior, as non-local types don't have leading digits. By that logic, we might even want to push down the stripping of leading digits into `DefaultTokenWriter.writeIdentifier`. We never want to print an identifier with leading digits, as that's not a valid Java identifier.

What do you think? The solution in here is not bad by any means, and is the preexisting one, but the repetition across methods just got me thinking.

Reply:
> but the repetition across methods just got me thinking

Yeah I thought about that too, especially as it's currently done in #4134 too - having it consistent in one place would make more sense. I'll look into it.

Reply:
calling the method in the DefaultTokenWriter fails due to https://github.com/INRIA/spoon/blob/a03c436e4f3b892cebc12d0a1b5ddad83e8e5c7c/src/main/java/spoon/reflect/visitor/DefaultJavaPrettyPrinter.java#L1883 writing the numbers of anonymous classes. As an example, the following test fails (the error is thrown in the stripLeadingDigits method, but the test would fail with an empty string too) and it's also expected to have a number there.

If that's the only case, I could just return the string itself instead of throwing the exception. This would make writing such stuff a bit slower but as it tends to be only a few chars (as long as you don't have hundreds of anonymous classes in one class) it shouldn't really matter.

Reply:
Test case you linked there is a pretty strange one. The method is called `testAnonymousClassesHaveAnEmptyStringForItsNameInNoClasspath`, the contract probably by accident states that all references should be empty strings, and the test then goes on to verify that the simple name of an anonymous class is a digit. And a bunch of _other_ fully qualified names. So, the test IMO is overspecified and hardly does what it says, but the question still remains: what should the fully qualified name of an anonymous class be, if it does not contain that digit?

Let's be pragmatic: the beautiful solution caused complications, let's go with the less beautiful one of doing `printer.writeIdentifier(stripLeadingDigits(ctEnum.getSimpleName());` all over the place :)

Reply:
> what should the fully qualified name of an anonymous class be, if it does not contain that digit?

As long as I didn't miss anything, there should be no case where printing the name of an anonymous type would be correct. So yeah, that's a good question.

I simplified the code now and it works with the tests that failed before, but it does not throw an exception anymore when writing anonymous type names as it seems to cause more headache than necessary I think.